dapps/shop/backend/utils/dns/clouddns.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * DNS utilities for GCP CLoud DNS
+ */
+
+const { DNS } = require('@google-cloud/dns')
+
+let CACHED_CLIENT
+const DEFAULT_TTL = 300 // 5 minutes
+
+/**
+ * Return a Google Cloud DNS API client
+ *
+ * @returns {DNS}
+ */
+function getClient(credentials) {
+  if (CACHED_CLIENT) return CACHED_CLIENT
+  if (!credentials) throw new Error('Must supply GCP credentails')
+  if (typeof credentials === 'string') credentials = JSON.parse(credentials)
+
+  CACHED_CLIENT = new DNS({ projectId: credentials.project_id, credentials })
+
+  return CACHED_CLIENT
+}
+
+/**
+ * Append e to s
+ *
+ * @param {string} s string we're appending to
+ * @param {string} e string we're appending
+ * @returns {string} s+e
+ */
+function append(s, e) {
+  if (s.endsWith(e)) return s
+  return `${s}${e}`
+}
+
+/**
+ * Looks for a record within a set of records matching name
+ *
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Record.html
+ *
+ * @param {Array} of Record
+ * @param {string} name of DNS record
+ * @returns {boolean} if a record matching name was found
+ */
+function containsRecord(records, name) {
+  for (let i=0; i<records.length; i++) {
+    if (records[i].name == name) return true
+  }
+  return false
+}
+
+/**
+ * Get a speific Zone
+ *
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Zone.html
+ *
+ * @param {string} name of DNS record
+ * @returns {Zone}
+ */
+async function getZone(DNSName) {
+  const dns = getClient()
+  const [zones] = await dns.getZones({ maxResults: 50 })
+
+  for (let i=0; i<zones.length; i++) {
+    const dnsName = zones[i].metadata.dnsName
+    if (dnsName == DNSName) {
```

Comment:
nit: as a convention, don't we use === ?

Reply:
Guess I got used to Python, thanks